### PR TITLE
Refactor firebase

### PR DESCRIPTION
**Pull request by anakhap**
Refactored the code for classes that use Firebase. The class FirebaseInstance was created in order to only have a single instance accessing Firebase. 

The following classes were modified: 
- EventCreation.java: creates a new event with the data from user input by passing it into addNewEventToDatabase() method. This method returns a boolean to ensure that the event was created and added to Firebase.
- Confirmation.java
- LoginActivity.java
- RegisterActivity.java
- ResetPassword.java
- SetUpProfile.java
- MainActivity.java
- UserProfileFragment.java

### CODE REVIEW

File: StudyBuddy/app/src/main/java/uw/studybuddy/Events/EventCreation.java [hunk 1]
```diff
@@ -35,7 +36,7 @@
     private TextView locationCreate;
     private TextView subjectCreate;
 
-    private DatabaseReference mDatabase;
+//    private DatabaseReference mDatabase;
```
Comment (Leksha) [inline StudyBuddy/app/src/main/java/uw/studybuddy/Events/EventCreation.java]: We can delete this now.

File: StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/Confirmation.java [hunk 2]
```diff
@@ -12,11 +12,12 @@
 import com.google.firebase.database.DatabaseReference;
 import com.google.firebase.database.FirebaseDatabase;
 
+import uw.studybuddy.FirebaseInstance;
 import uw.studybuddy.R;
 
 public class Confirmation extends AppCompatActivity {
 
-    private FirebaseAuth mAuth;
+    //private FirebaseAuth mAuth;
```
Comment (Leksha) [inline StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/Confirmation.java]: Can be deleted

File: StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/Confirmation.java [hunk 3]
```diff
@@ -27,15 +28,15 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_confirmation);
 
-        mAuth = FirebaseAuth.getInstance();
+        //mAuth = FirebaseAuth.getInstance();
```
Comment (Leksha) [inline StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/Confirmation.java]: Can be deleted

File: StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/LoginActivity.java [hunk 4]
```diff
@@ -19,20 +19,23 @@
...
+import com.google.firebase.auth.FirebaseUser;
 
+import uw.studybuddy.FirebaseInstance;
 import uw.studybuddy.MainActivity;
 import uw.studybuddy.R;
 
 public class LoginActivity extends AppCompatActivity{
 
-    private FirebaseAuth mAuth;
+    //private FirebaseAuth mAuth;
```
Comment (Leksha) [inline StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/LoginActivity.java]: Can be deleted

File: StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/LoginActivity.java [hunk 5]
```diff
@@ -19,20 +19,23 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         setTheme(R.style.AppTheme_NoActionBar);
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
 
-        mAuth = FirebaseAuth.getInstance();
+        //mAuth = FirebaseAuth.getInstance();
```
Comment (Leksha) [inline StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/LoginActivity.java]: Can be deleted

File: StudyBuddy/app/src/main/java/uw/studybuddy/FirebaseInstance.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        return retVal;
+
+    }
+
+    public static FirebaseAuth registerConfirmation() {
+        FirebaseAuth curUser = getFirebaseAuthInstance();
+        return curUser;
+    }
```
Comment (Leksha) [inline StudyBuddy/app/src/main/java/uw/studybuddy/FirebaseInstance.java]: Does this method not just return `mAuth` which is done by the  `getFirebaseAuthInstance()`?
By this I mean if we need this method or can we just call `getFirebaseAuthInstance` instead?

File: StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/LoginActivity.java [hunk 7]
```diff
@@ -19,20 +19,23 @@
...
+        final FirebaseUser user = FirebaseInstance.registerConfirmation().getCurrentUser();
```
Comment (Leksha) [inline StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/LoginActivity.java]: Can we instead call `FirebaseInstance.getFirebaseAuthInstance().getCurrentUser()` ?

File: StudyBuddy/firebase_library/src/androidTest/java/com/example/firebase_library/ExampleInstrumentedTest.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
+package com.example.firebase_library;
```
Comment (Leksha) [inline StudyBuddy/firebase_library/src/androidTest/java/com/example/firebase_library/ExampleInstrumentedTest.java:1]: Can we find a way to delete this library?

**Review by Leksha — CHANGES_REQUESTED** [review]
I haven't tested the scenarios yet but this looks great!

File: StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/LoginActivity.java (hunk 7 above)
Comment (anakhap) [reply StudyBuddy/app/src/main/java/uw/studybuddy/LoginAndRegistration/LoginActivity.java]: That instance (user) is not actually used. The snippet of code right under it was using it, but its been commented out for now. Should I just remove it?

File: StudyBuddy/app/src/main/java/uw/studybuddy/FirebaseInstance.java (hunk 6 above)
Comment (anakhap) [reply StudyBuddy/app/src/main/java/uw/studybuddy/FirebaseInstance.java]: True, you are right this is redundant. I will remove it.

**Review by Leksha — APPROVED** [review]

**Pull request merged**